src/components/post-card/PostCard.js

```diff
@@ -1,148 +1,181 @@
+    //   const { post: nextPost } = nextProps;
+    //
+    //   return curPost.id !== nextPost.id
+    // }
+
+    render() {
+        const {post, hasImage, author = '', className = ''} = this.props;
+        const {title, body} = post;
+        {/* todo с помощью  деструктуризации достать из this.state проперти showComments, error, isCommentsLoading, commentsLoaded */
         }
-        {
-          //todo если секция комментариев открыта, т.е. showComments = true
-          //   но НЕ идет загрузка комментариев, т.е. isCommentsLoading = false
-          //   и запрос уже был выполнен т.е. commentsLoaded = true
-          //   и массив comments пустой, т.е. !comments.length
-          // показываем сообщение, что нет результатов (div с надписью "No comments for this post yet" в div)
-        }
-        {
-          // todo если секция комментариев открыта, т.е. showComments = true
-          //   и НЕ идет загрузка комментариев, т.е. isCommentsLoading = false
-          //   и запрос уже был выполнен т.е. commentsLoaded = true
-          //   и массив comments НЕ пустой, т.е. !!comments.length
-          //    то:
-          comments.map(comment => (<Comment comment={comment} key={comment.id} />))
-        }
-      </div>
-    );
-  }
+        const {comments, showComments, error, isCommentsLoading, commentsLoaded} = this.state;
+
+        const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+
+        console.log('PostCard render');
+        return (
+            <div className={`may-post-card card ${className}`}>
+                <div className="may-post-card-img" id="my-block" onClick={() => {
+                    alert('ghvcdhfvbdfsjvbdf');
+                }}>
+                    <img src={hasImage ? kittyUrl : DefaultImg}/>
+                </div>
+                <div className="card-body">
+                    <h4 className="card-title title">{title}</h4>
+                    <div className="card-text body">
+                        {body}
+                    </div>
+                    <blockquote className="blockquote">
+                        <footer className="blockquote-footer">Author:
+                            <cite title="Source Title">{author}</cite>
+                        </footer>
+                    </blockquote>
+                </div>
+
+                {
+                    // todo: добавить label, в котором как children будет если showComments = true - надпись "Hide comments",
+                    //    иначе - "Show comments"
+                    //    повесить на нее onClick событие this.onToggleComments
+                    //    как класс задать ей "btn btn-link"
+                }
+                {
+                    showComments
+                    ? <label className="btn btn-link" onClick={this.onToggleComments}>Hide comments</label>
+                    : <label className="btn btn-link" onClick={this.onToggleComments}>Show comments</label>
```

Comment:
Nice!
also, you can use only 1 label tag and render different string inside it using same ternary operator